pyrefly/lib/module/ignore.rs

```diff
@@ -71,36 +76,61 @@ impl Ignore {
     }
 
     pub fn get_suppression_kind(line: &str) -> Option<SuppressionKind> {
-        fn match_pyrefly_ignore(line: &str) -> bool {
+        fn match_pyrefly_ignore(line: &str) -> Option<SuppressionKind> {
             let mut words = line.split_whitespace();
             if let Some("pyrefly:") = words.next() {
-                words.next() == Some("ignore")
-            } else {
-                false
+                if let Some(word) = words.next() {
+                    if word == "ignore" {
+                        return Some(SuppressionKind::Pyrefly);
+                    }
+
+                    if word.starts_with("ignore[") && word.ends_with(']') {
+                        if let Ok(kind) = ErrorKind::from_str(&word[7..word.len() - 1]) {
```

Comment:
minor nit, feel free to ignore if you disagree, but can we make the index this instead?

```
&word["ignore[".len()..word.len() - 1]
```

Reply:
Pretty much any string indexing is a poor plan. Better to do `let Some(word) = word.strip_prefix("ignore[") && let Some(word) = word.strip_suffix(']')` and then you don't have to muck around with string offsets.